frontend/lib/src/components/core/Block/ElementNodeRenderer.tsx

```diff
@@ -484,6 +476,22 @@ const RawElementNodeRenderer = (
       )
     }
 
+    case "arrowVegaLiteChart":
+      const vegaLiteElement = node.vegaLiteChartElement as VegaLiteChartElement
+      return (
+        <ArrowVegaLiteChart
+          element={vegaLiteElement}
+          // Vega-lite chart can be used as a widget (when selections are activated) or
+          // an element. We only want to set the key in case of it being used as a widget.
```

Comment:
why can't we always set a key; is there a disadvantage? Because we don't have enough information to generate it?

Reply:
Alright, I see. I think it would be good to add a short form of this explanation as a comment into the code, so that the next one who will look at this has some context.

Reply:
Update the comment